util/storage-samples/storage-owner-sample/src/main/java/org/fidoalliance/fdo/storage/OwnerDbManager.java

```diff
@@ -330,11 +333,20 @@ public void updateMtu(DataSource ds, String field, int mtu) {
 
     String sql = "UPDATE TO2_SETTINGS" + " SET " + field + " = ?" + " WHERE ID = 1;";
 
-    if (mtu > 0 && mtu < Const.SERVICE_INFO_MTU_MIN_SIZE) {
-      System.out.println(
-          "Received value less than default minimum of 256 bytes. "
-              + "Updating MTU size to default minimum");
-      mtu = Const.SERVICE_INFO_MTU_MIN_SIZE;
+    if (field.equals("DEVICE_SERVICE_INFO_MTU_SIZE")) {
+      if (mtu > 0 && mtu < Const.SERVICE_INFO_MTU_MIN_SIZE) {
+        System.out.println("Received value less than default minimum. "
+                + "Updating MTU size to default minimum of "
+                + Const.SERVICE_INFO_MTU_MIN_SIZE);
+        mtu = Const.SERVICE_INFO_MTU_MIN_SIZE;
+      }
+      // check max limit as well
+      if (mtu > 0 && mtu > Const.OWNER_THRESHOLD_DEFAULT_MTU_SIZE) {
+        System.out.println("MTU size greater than maximum allowed. "
+                 +  "Updating MTU size to maximum limit of "
+                 +  Const.OWNER_THRESHOLD_DEFAULT_MTU_SIZE);
+        mtu = Const.SERVICE_INFO_MTU_MIN_SIZE;
+      }
     }
 
```

Comment:
Consider an MTU value of -100, we won't be entering the `if block`  neither at ln 337 nor  344 and the negative value gets updated in the database(ln 354). I think you can remove `mtu > 0` check or add an additional check 
`if ( mtu < 0 ) { mtu = Const.SERVICE_INFO_MTU_MIN_SIZE; }`

Reply:
Updated logic